metricbeat/module/golang/expvar/expvar.go

```diff
@@ -85,13 +83,10 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // Fetch methods implements the data gathering and data conversion to the right format
 // It returns the event which is then forward to the output. In case of an error, a
 // descriptive error must be returned.
-func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
+func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
 	json, err := m.http.FetchJSON()
-
 	if err != nil {
-		logger.Error(err)
-		reporter.Error(err)
-		return
+		return errors.Wrap(err, "Error in http fetch")
```

Comment:
Nit: errors inside errors should start lower case. I wonder why Hound did not complain.

Reply:
Oddly enough, the linter I have in vscode usually complains about that as well. Might be more context-dependent?